incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -246,15 +246,15 @@ func (v *Hierarchy) getConflictingObjectsOfType(gvk schema.GroupVersionKind, new
 	// Get all the source objects in the new ancestors that would be propagated
 	// into the descendants.
 	newAnsSrcObjs := make(map[string]bool)
-	for _, o := range newParent.GetPropagatingObjects(gvk) {
+	for _, o := range newParent.GetAncestorSourceObjects(gvk, "") {
 		newAnsSrcObjs[o.GetName()] = true
 	}
```

Comment:
TODO (maybe for Ginny): this doesn't obey the shouldPropagate rules, but it probably should (and will definitely have to once we have exceptions).